Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift

```diff
@@ -212,34 +212,8 @@ extension DefaultPublisher {
 
     // MARK: Track
     private func performTrackTrackableEvent(_ event: Event.TrackTrackableEvent) {
-        guard !state.isStoppingOrStopped else {
-            publisherStoppedCallback(handler: event.resultHandler)
-            return
-        }
-
-        guard !trackables.contains(event.trackable) else {
-            enqueue(event: .trackableReadyToTrack(.init(trackable: event.trackable, resultHandler: event.resultHandler)))
-            return
-        }
-
-        ablyPublisher.connect(
-            trackableId: event.trackable.id,
-            presenceData: presenceData,
-            useRewind: false
-        ) { [weak self] result in
-            switch result {
-            case .success:
-                self?.enqueue(event: .presenceJoinedSuccessfully(.init(trackable: event.trackable) { [weak self] result in
-                    switch result {
-                    case .success:
-                        self?.enqueue(event: .trackableReadyToTrack(.init(trackable: event.trackable, resultHandler: event.resultHandler)))
-                    default:
-                        return
-                    }
-                }))
-            case .failure(let error):
-                self?.callback(error: error, handler: event.resultHandler)
-            }
+        performAddOrTrack(event.trackable, resultHandler: event.resultHandler) {
```

Comment:
We need to add `[weak self] in` to avoid memory leaks. So this line should be:
```
        performAddOrTrack(event.trackable, resultHandler: event.resultHandler) { [weak self] in
```

Reply:
02def08355771db5938ed30dae6879a75c88b657

Reply:
You'll still need to add a `?` after self, since now you have a weak reference to `self` here:
```
            self?.enqueue(event: .trackableReadyToTrack(.init(trackable: event.trackable, resultHandler: event.resultHandler)))
```
I imagine it's not compiling currently ;)

Reply:
feb724012d3234dceea5ebbb18e3d84fd00ba23b